prompting/tasks/benchmark.py

```diff
@@ -0,0 +1,109 @@
+import re
+import bittensor as bt
+from dataclasses import dataclass
+from prompting.tasks import Task
+from prompting.utils.exceptions import TaskCreationError
+
+# TODO: introduce criteria for the query and reference answer (length, layout, etc.) and make these arguments
+
+# Used to instruct the LLM to provide a good query when given a context
+QUERY_SYSTEM_PROMPT = """\
+You are a multiple choice question-generating expert, focusing on delivering comprehensive and accurate questions with depth and clarity for AI benchmarking purposes. The questions you generate should be specific and based on the context that is provided. Do not add any greetings or assistant messages before or after the question (e.g. Here's a question based on the context) as this does not bring any benefit to the question. The question should be self-contained. You will adhere to a word limit of 100 words for each question.
+
+In addition to the question, you will provide 4 possible answers (multiple choice-style), one of which is correct. Ensure that the correct answer is not too obvious, and that the incorrect answers are plausible. Be careful to maintain a neutral tone in the candidate answers, and randomly order the answers so that the correct answer is not always in the same position. Do not add any additional information after the answers.
```

Comment:
Good call. From the tracked experiment, 4/209 of the queries had a repeated option.

```
In [132]: df['query'].apply(lambda text: len(set(re.findall(r"\n\*?[A-D]\.\s+(.*)", text)))).value_counts()
Out[132]: 
query
4    205
3      4
Name: count, dtype: int64
```

Reply:
We can add both an extra instruction and a further acceptance criteria which won't allow these ones to be used.

Reply:
For such low rates I would recommend checking whether a duplicate answer exists and simply regenerating rather than changing the prompt. Overall, that ends up somewhat cheaper and gives you 100% confidence that now answer is ever repeated. You can of course still modify the prompt slightly for extra safety, but with LLMs the more hard checks you can do the better, prompt engineering is always somewhat of a mess.